net/graphsync_fetcher_test.go

```diff
@@ -88,14 +98,232 @@ func TestGraphsyncFetcher(t *testing.T) {
 			return false, nil
 		}
 
-		ts, err := fetcher.FetchTipSets(ctx, final.Key(), peer.ID("fake"), done)
+		ts, err := fetcher.FetchTipSets(ctx, final.Key(), pid0, done)
 		require.NoError(t, err, "the request completes successfully")
 		require.Equal(t, 4, len(mgs.receivedRequests), "all expected graphsync requests are made")
 		require.Equal(t, 2, len(ts), "the right number of tipsets is returned")
 		require.True(t, final.Key().Equals(ts[0].Key()), "the initial tipset is correct")
 		require.True(t, gen.Key().Equals(ts[1].Key()), "the remaining tipsets are correct")
 	})
 
+	t.Run("initial request fails on a block but fallback peer succeeds", func(t *testing.T) {
+		gen := builder.NewGenesis()
+		final := builder.AppendOn(gen, 3)
+		height, err := final.Height()
+		require.NoError(t, err)
+		chain1 := types.NewChainInfo(pid1, final.Key(), height)
+		chain2 := types.NewChainInfo(pid2, final.Key(), height)
+		pt := &fakePeerTracker{[]*types.ChainInfo{chain1, chain2}}
+
+		stubs := []requestResponse{
```

Comment:
what is this a fake request for / response from?

Reply:
this would be a request to another peer with graphsync and a response from that peer.